cmd/plugin/plugin.go

```diff
@@ -29,39 +29,88 @@ func NewCommand(out output.Output) *cobra.Command {
 
 // InstallCommand creates a new command to install
 func InstallCommand(out output.Output) *cobra.Command {
-	return &cobra.Command{
+	var (
+		pluginSource  string
+		pluginName    string
+		pluginVersion string
+		pluginURL     string
+		avm           avmpkg.AVM
+	)
+
+	cmd := &cobra.Command{
 		Use:   "install",
 		Short: "Installs a tool for a specific plugin",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			avm, err := avmpkg.New(out)
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if pluginSource != "asdf" {
+				return fmt.Errorf("We do not support any source other than asdf for now, got: %s", pluginSource)
+			}
+
+			if pluginName == "" {
+				return fmt.Errorf("Please provide a plugin name")
+			}
+
+			var err error
+			avm, err = avmpkg.New(out)
 			if err != nil {
 				return fmt.Errorf("failed to initialize avm: %w", err)
 			}
 
+			// We only need to check for existing plugins if we do not get a pluginURL
+			if pluginURL == "" {
+				_, err = avm.GetDefaultSource().GetPlugin(pluginName)
+				if err != nil {
+					return err
+				}
+			}
```

Comment:
?? why do we need this check?

Reply:
I wanted to check if a plugin is available and only do the install work if it is. Should we remove it?